packages/apple-pay-integration/e2e/ApplePaySessionPaymentStepMockObject.ts

```diff
@@ -0,0 +1,137 @@
+            console.log('completeShippingMethodSelection', _update);
+        }
+
+        completeMerchantValidation(_response) {
+            console.log('completeMerchantValidation', _response);
+        }
+
+        completePaymentMethodSelection(newTotal: ApplePayJS.ApplePayLineItem, newLineItems: ApplePayJS.ApplePayLineItem[]): void;
+        completePaymentMethodSelection(update: ApplePayJS.ApplePayPaymentMethodUpdate): void;
+
+        completePaymentMethodSelection(newTotal: unknown, newLineItems?: unknown): void {
+            console.log('completeMerchantValidation', newTotal, newLineItems);
+        }
+        begin() {
+            setTimeout(() => {
+                const event = {
+                    validationURL: 'https://www.example.com',
+                } as ApplePayJS.ApplePayValidateMerchantEvent
+
+                this.onvalidatemerchant(event);
+            }, 0);
+            setTimeout(() => {
+                const mockPaymentData = {
+                    version: "mock_v1",
+                    data: "mockData",
+                    signature: "mockSignature",
+                    header: {
+                        ephemeralPublicKey: "mockPublicKey",
+                        publicKeyHash: "mockPublicKeyHash",
+                        transactionId: "mockTransactionId",
+                    }
+                }
+                const event = {
+                    payment: {
+                        token: {
+                            paymentData: mockPaymentData,
+                            paymentMethod: {
+                                displayName: 'xx',
+                                network: 'xx',
+                                type: 'debit',
+                                paymentPass: {
+                                    primaryAccountIdentifier: 'xx',
+                                    primaryAccountNumberSuffix: 'xx',
+                                    activationState: 'activated'
+                                }
+                            },
+                            transactionIdentifier: "xx",
+                        },
+                        billingContact: {
+                            emailAddress: "mock@mock.com",
+                            familyName: "mock",
+                            givenName: "mock",
+                            phoneNumber: "00000000",
+                        },
+                        shippingContact: {
+                            emailAddress: "mock@mock.com",
+                            familyName: "mock",
+                            givenName: "mock",
+                            phoneNumber: "00000000",
+                        }
+                    },
+                } as unknown as ApplePayJS.ApplePayPaymentAuthorizedEvent;
+
+                this.onpaymentauthorized(event);
+            }, 1000)
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
Do you still need `ts-ignore`?

Reply:
Not needed anymore, will remove for this file